src/cls/rbd/cls_rbd_types.h

```diff
@@ -133,6 +133,7 @@ inline void decode(MirrorImageStatusState &state, bufferlist::iterator& it)
 
 struct MirrorImageStatus {
   MirrorImageStatus() {}
+  ~MirrorImageStatus() {}
```

Comment:
This should be virtual IIUC since MirrorImageStatus is a "base class" for MirrorImageStatusOnDisk and I suspect somewhere we are deleting MirrorImageStatusOnDisk objects via a MirrorImageStatus* and that is what triggers the error in clang. I'll ask for a second opinion on this though. @tchaikov would you mind giving your opinion here?